log/zap/field.go

```diff
@@ -59,3 +61,158 @@ func (t trace) MarshalLogObject(enc zapcore.ObjectEncoder) error {
 	enc.AddBool("sampled", j.IsSampled())
 	return nil
 }
+
+type spanContext struct {
+	spanContext jaeger.SpanContext
+}
+
+// Context creates a zap.Field which marshals all information contained in a jaeger context
+func Context(sc jaeger.SpanContext) zapcore.Field {
+	return zap.Object("context", spanContext{spanContext: sc})
+}
+
+func (s spanContext) MarshalLogObject(enc zapcore.ObjectEncoder) error {
+	ctx := s.spanContext
+	enc.AddString("trace", ctx.TraceID().String())
+	enc.AddString("span", ctx.SpanID().String())
+	enc.AddString("parent", ctx.ParentID().String())
+	enc.AddBool("debug", ctx.IsDebug())
+	enc.AddBool("sampled", ctx.IsSampled())
+	enc.AddBool("firehose", ctx.IsFirehose())
+	s.encodeBaggage(ctx, enc)
+	return nil
+}
+
+func (s spanContext) encodeBaggage(ctx jaeger.SpanContext, enc zapcore.ObjectEncoder) {
+	var baggage baggageKVs
+	ctx.ForeachBaggageItem(func(k, v string) bool {
+		baggage = append(baggage, baggageKV{
+			key:   k,
+			value: v,
+		})
+		return true
+	})
+
+	enc.AddArray("baggage", baggage)
```

Comment:
it seems this ends up calling `(b baggageKVs) MarshalLogArray()`. Wonder if you could do this without creating an intermediate array `baggageKVs`, by simply subtyping SpanContext one more time, e.g. `type baggageKVs SpanContext`, and then doing `ForeachBaggageItem` inside `(b baggageKVs) MarshalLogArray()`. The only gotcha if zap would check for the argument to AddArray to be actually an array (which it has no reason to do, apparently).

Reply:
Good suggestion - I was able to remove the intermediate array!